src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -513,9 +515,10 @@ void recordWriteAfterQueryExecution() throws SQLException {
 
     public Statement getReadStatement(String sql) {
         isWriteOperation = isFunctionCall(sql) || isUpdate(sql) || isDelete(sql);
+
         // if write connection is already initialized, but the current statement is null
         // we should use a write statement, regardless of the fact that readStatement has been called.
-        if (connectionProvider.hasWriteConnection() || isWriteOperation || isSelectForUpdate(sql)) {
+        if (connectionProvider.getState().equals(MAIN) || isWriteOperation || isSelectForUpdate(sql)) {
```

Comment:
Right now `ReplicaStatement` need to know about MAIN, what increase coupling, and I think it would be better to have two different methods

1. `hasWriteConnection`
2. `hasOnlyWriteConnection`

Reply:
> Right now ReplicaStatement need to know about MAIN

The state is part of the API. What's wrong in `knowing` it?